src/client/lib/hooks/useRecaptcha.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export type RenderOptions = {
+  sitekey: string;
+  size?: string;
+  callback: (token: string) => void;
+  'error-callback': (value: undefined) => void;
+  'expired-callback': (value: undefined) => void;
+};
+
+export const recaptchaReady = () =>
+  typeof window !== 'undefined' &&
+  typeof window.grecaptcha !== 'undefined' &&
+  typeof window.grecaptcha.render === 'function';
+
+const useRecaptchaScript = (
+  url = 'https://www.google.com/recaptcha/api.js?render=explicit',
+) => {
+  const [loaded, setLoaded] = React.useState(false);
+
+  React.useEffect(() => {
+    const scriptExists =
+      typeof document !== undefined &&
+      document.getElementById('g-captcha-script');
+
+    if (scriptExists || recaptchaReady()) {
+      setLoaded(true);
+      return;
+    }
+    const script = document.createElement('script');
```

Comment:
would be good to use `loadScript` from https://github.com/guardian/libs here if possible?

Reply:
interesting - I wasn't aware that this existed. I'll look into it :)